discord_bot/discord_bot.py

```diff
@@ -144,24 +142,45 @@ def load_config(config_file_path):
                     self.logger.warning("Invalid Session")
                 elif message["op"] == Opcodes.DISPATCH:
                     event = message["t"]
-                    if event == Events.READY:
+                    if event == Events.READY.value:
                         self.session_id = message["d"]["session_id"]
+                    elif event == Events.MESSAGE_CREATE.value:
+                        channel_id = message["d"]["channel_id"]
+                        author = message["d"]["author"]["username"]
+                        content = message["d"]["content"]
+                        await self.respond_message(channel_id, author, content)
                 else:
                     self.logger.exception("Unexpected opcode {}: {}".format(message["op"], message))
 
     # TODO: Remove this test code
-    # async def send_message(self, recipient_id, content):
-    #     """
-    #     Post a message into the given channel
-    #     :param recipient_id: num - the recipient to open a DM channel with
-    #     :param content: obj - message sent
-    #     :return: dict - Fires a "Message Create" Gateway event
-    #     """
-    #     channel = requests.post("{}", json={"recipient_id": recipient_id}).json()
-    #
-    #     return requests.post(
-    #         "{}/channels/{}/messages".format(
-    #             self.config["discord_api_endpoint"], channel["id"]),
-    #         json={
-    #             "content": content
-    #         }).json()
+    async def respond_message(self, channel_id, author, content):
+        """
+        Receives a message and responds to the user in the same channel.
+        :param channel_id: int - channel id
+        :param author: string - author of the message
+        :param content: string - message the user inputs
+        :return:
+        """
+        defaults = {
```

Comment:
This is incredibly confusing and also extremely weird. You're defining a dictionary, then redefining another dictionary based on it partially with keyword arguments and partially copying another dictionary, and then passing it as a different keyword argument just to make it more confusing.

I heavily advise you to never use `kwargs` unless you know exactly what you're doing and why you need to do it; you should be explicit with your arguments.

Here's an example of how it should look:
http://aiohttp.readthedocs.io/en/stable/client.html#custom-headers